googlespreadsheet/googlespreadsheet-connector/googlespreadsheet-connector-2.0.0/org.wso2.carbon.connector/src/main/resources/config/init.xml

```diff
@@ -17,30 +17,64 @@
   ~ under the License.
   -->
 <template name="init" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter name="refreshToken" description="Refresh Token"/>
+    <parameter name="clientId"
+               description="clientId of your App, given by google console when you registered your application"/>
+    <parameter name="clientSecret"
+               description="clientSecret of your App, given by google console when you registered your application"/>
+    <parameter name="apiUrl" description="API Url to access the endpoint."/>
     <parameter name="accessToken" description="Access Token if a client account is being used"/>
+    <parameter name="accessTokenRegistryPath" description="API Url to access the endpoint."/>
     <parameter name="apiVersion" description="The Googlespreadsheet API request version."/>
     <sequence>
-        <property name="uri.var.accessToken" expression="$func:accessToken" scope="default"
-                  type="STRING"/>
-        <property name="uri.var.apiVersion" expression="$func:apiVersion" scope="default"
-                  type="STRING"/>
-        <property name="uri.var.apiUrl" value="https://spreadsheets.google.com"/>
+        <property name="uri.var.refreshToken" expression="$func:refreshToken"/>
+        <property name="uri.var.clientId" expression="$func:clientId"/>
+        <property name="uri.var.clientSecret" expression="$func:clientSecret"/>
+        <property name="uri.var.apiUrl" expression="$func:apiUrl"/>
+        <property name="uri.var.accessToken" expression="$func:accessToken"/>
+        <property name="uri.var.accessTokenRegistryPath" expression="$func:accessTokenRegistryPath"/>
         <property name="uri.var.apiVersion" expression="$func:apiVersion"/>
         <property name="uri.var.appVersionId" value=""/>
-
         <filter xpath="(not(get-property('uri.var.apiVersion') = '' or (not(string(get-property('uri.var.apiVersion'))))))">
             <then>
                 <property name="uri.var.appVersionId"
                           expression="fn:concat(get-property('uri.var.appVersionId'),'v=',get-property('uri.var.apiVersion'))"/>
             </then>
         </filter>
-        <header name="Authorization"
-                expression="fn:concat('Bearer ',get-property('uri.var.accessToken'))"
+        <filter source="boolean($ctx:uri.var.accessTokenRegistryPath)" regex="false">
+            <then>
+                <property name="uri.var.accessTokenRegistryPath"
+                          value="connectors/Google/accessToken"/>
+            </then>
+            <else>
+                <property name="uri.var.accessTokenRegistryPath"
+                          expression="$ctx:uri.var.accessTokenRegistryPath"/>
+            </else>
+        </filter>
+        <filter source="boolean($ctx:uri.var.accessToken)" regex="false">
```

Comment:
Change to following please,

<property name="uri.var.accessToken"
                          expression="get-property('registry', $ctx:uri.var.accessTokenRegistryPath)"/>
        <filter source="boolean($ctx:uri.var.accessToken)" regex="false">
               <then>
                     <property name="uri.var.accessToken" expression="$func:accessToken"/>
               </then>
        </filter>